### ISSUE TITLE
"version" and "version_id" ignored in [download ...], works in [download_data ...]

### ISSUE THREAD

**Issue by FreddieChopin** [download-monitor/download-monitor#344]
I've created a download with two files with different versions, "1" (.zip file) and "2" (.7z file). Then I've created a page with following content:

```
[download id="138" template="filename" version="1"]
Size: [download_data id="138" data="filesize" version="1"]
Name: [download_data id="138" data="filename" version="1"]
Version: [download_data id="138" data="version" version="1"]
Date: [download_data id="138" data="file_date" version="1"]
SHA1: [download_data id="138" data="sha1" version="1"]

[download id="138" template="filename" version="2"]
Size: [download_data id="138" data="filesize" version="2"]
Name: [download_data id="138" data="filename" version="2"]
Version: [download_data id="138" data="version" version="2"]
Date: [download_data id="138" data="file_date" version="2"]
SHA1: [download_data id="138" data="sha1" version="2"]
```

The problem is that the data displayed by `[download_data ...]` shortcode does relate to selected version, but the link generated by `[download ...]` is always for the "blue" (latest) version. The same thing is true if I use "version_id" (with proper IDs instead of version names).

Here's the content of generated page:

> distortos-master.zip	(0 downloads) 
> Size: 2 MB
> Name: distortos-master.zip
> Version: 1
> Date: February 22, 2016
> SHA1: db9bf773a04b6b7c45634425171d2f8c3dcc3584
> 
> distortos-master.zip	(0 downloads) 
> Size: 397 kB
> Name: distortos-master.7z
> Version: 2
> Date: February 22, 2016
> SHA1: 8ff5ce4903053e09787e70b6d8885070ebf26fef

Am I doing something wrong or is this a bug?

Latest plugin version with latest Wordpress.

**Comment by FreddieChopin** [comment]
I think the problem is caused by this line - https://github.com/download-monitor/download-monitor/blob/c59c1fd0022239edf67e84395cd619d65921fc98/includes/class-dlm-shortcodes.php#L139 Even though the version_id is set a few lines above, template handler gets a new instance of the DLM_Download object which is constructed with just the download ID (not the version ID).

**Comment by barrykooij** [comment]
Confirmed, thanks for reporting @FreddieChopin

**Comment by barrykooij** [comment]
Fixed at 3ca5525d0d4d02e207bc8d759eefefc078ec354a
